app/announcement/services/announcement_service.py

```diff
@@ -41,3 +49,95 @@ def create(self, create: AnnouncementCreateBodyPayload) -> AnnouncementView:
+        matched, match_score = self.__get_matched_items(filters, true_items)
+
+        m = len(matched)
+        n = len(filters)
+
+        announcement_coordinates = (announcement.address.latitude, announcement.address.longitude)
+        d = haversine(UFCG_COORDINATES, announcement_coordinates)
+
+        score = ((m / n) * match_score) + (RADIUS - (d / RADIUS))
+
+        return score
+
+    def __extract_true_items(self, announcement: AnnouncementView):
+
+        true_items = list()
+
+        if announcement.allow_events:
+            true_items.append(AnnouncementTagsEnum.ALLOW_EVENTS)
+        if announcement.allow_pet:
+            true_items.append(AnnouncementTagsEnum.ALLOW_PETS)
+        if announcement.has_furniture:
+            true_items.append(AnnouncementTagsEnum.HAS_FURNITURE)
+        if announcement.has_internet:
+            true_items.append(AnnouncementTagsEnum.HAS_INTERNET)
+        if announcement.has_piped_gas:
+            true_items.append(AnnouncementTagsEnum.HAS_PIPED_GAS)
+        if announcement.is_close_to_supermarket:
+            true_items.append(AnnouncementTagsEnum.IS_CLOSE_TO_SUPERMARKET)
+        if announcement.is_close_to_university:
+            true_items.append(AnnouncementTagsEnum.IS_CLOSE_TO_UNIVERSITY)
+
+        for vacancy in announcement.vacancies:
+            if vacancy.allowed_smoker:
+                true_items.append(AnnouncementTagsEnum.ALLOWED_SMOKER)
+            if vacancy.has_bathroom:
+                true_items.append(AnnouncementTagsEnum.HAS_BATHROOM)
+            if vacancy.has_cable_internet:
+                true_items.append(AnnouncementTagsEnum.HAS_CABLE_INTERNET)
+            if vacancy.is_shared_room:
+                true_items.append(AnnouncementTagsEnum.IS_SHARED_ROOM)
+            if vacancy.required_extroverted_person:
+                true_items.append(AnnouncementTagsEnum.REQUIRED_EXTROVERTED_PERSON)
+            if vacancy.required_organized_person:
+                true_items.append(AnnouncementTagsEnum.REQUIRED_ORGANIZED_PERSON)
+
+            if vacancy.gender == GenderEnum.FEMALE:
+                true_items.append(AnnouncementTagsEnum.FEMALE_GENDER)
+            elif vacancy.gender == GenderEnum.MALE:
+                true_items.append(AnnouncementTagsEnum.MALE_GENDER)
+
+        return true_items
+
+    def __get_matched_items(self, filters, items):
+        matched = list()
+        match_score = 0
+
+        for i in range(len(filters)):
+            filter = filters[i]
+            if filter in items:
+                matched.append(filter)
+                match_score += i
+
+        if (
+            (AnnouncementTagsEnum.MALE_GENDER in filters)
+            and (AnnouncementTagsEnum.MALE_GENDER not in matched)
+        ) or (
+            (AnnouncementTagsEnum.FEMALE_GENDER in filters)
+            and (AnnouncementTagsEnum.FEMALE_GENDER not in matched)
+        ):
+            match_score = 0
```

Comment:
Faz sentido a gente zerar? Não seria melhor a gente fazer uma dedução percentual do (pontuação)? Porque digo isso? Se o match for praticamente perfeito, porém o gênero diferir, o anúncio vai basicamente para o fim da fila.

Reply:
Falei com Igor ontem que o gênero talvez seja uma característica que tenha um peso bem grande, pois meio que mesmo se o match for quase perfeito a pessoa não poderia ocupar a vaga (em tese), já que a vaga é exclusiva para pessoas de um gênero que não é o dela, por isso pensei em zerar essa parcela do match, nesses casos o que vai pesar é só a distância entre o campus e a residência do anúncio. Mas podemos repensar isso, talvez deixar isso fora da nossa responsabilidade e deixar que esses detalhes se resolvam na negociação entre anunciante e interessando mesmo.

Reply:
Acho que vale a gente aplicar uma penslização alta, mas não zerar. 50% me parece razoavel

Reply:
Acho massa uns 50%, posso fazer essa mudança agora

Reply:
Vou só atualizar os testes pra essa mudança de cálculo